core/store/migrate/migrations/0195_add_evm_chain_id_to_job_specs.go

```diff
@@ -0,0 +1,73 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pressly/goose/v3"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/chainlink"
+)
+
+func init() {
+	goose.AddMigrationContext(Up195, Down195)
+}
+
+const (
+	up195_1 = `
+	UPDATE direct_request_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE flux_monitor_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE ocr_oracle_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE keeper_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE vrf_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE blockhash_store_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	UPDATE block_header_feeder_specs SET evm_chain_id = '%d' WHERE evm_chain_id IS NULL;
+	`
+	up195_2 = `
+	ALTER TABLE direct_request_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE flux_monitor_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE ocr_oracle_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE keeper_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE vrf_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE blockhash_store_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	ALTER TABLE block_header_feeder_specs ALTER COLUMN evm_chain_id SET NOT NULL;
+	`
+
+	down195 = `
+	ALTER TABLE direct_request_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE flux_monitor_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE ocr_oracle_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE keeper_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE vrf_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE blockhash_store_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+	ALTER TABLE block_header_feeder_specs ALTER COLUMN evm_chain_id DROP NOT NULL;
+    `
+)
+
+// nolint
+func Up195(ctx context.Context, tx *sql.Tx) error {
+	var opts chainlink.GeneralConfigOpts
+	cfg, err := opts.New()
```

Comment:
Does this have side effects?

Reply:
If you are talking about the whole migration, I don't think so. It should just fill any empty evm chainID fields in old job specs. New job specs can't be created without specifying evmChainID.

If you mean config loading because comment is left on that line, then I am not sure what kind of side effects there might be. Also I'm not fully sure how env vars are passed around for loading config when migrations are run, might have to inject the config somehow if it doesn't get loaded here by itself.

Reply:
Since NOPs  currently only run nodes on one chain at a time (other than CCIP which isn't affected by this), pulling evm chain id from toml config and setting it to job specs makes sense to me. Unless specs tables somehow stored specs that belong to multiple different chains, but that doesn't sound like it can happen to me?

Reply:
I just mean the config. The `New()` method looks clean though.